cosmos-gui/src/mysql_driver.js

```diff
@@ -37,89 +37,99 @@ var connection = mysql.createConnection({
     database: mysqlConfig.database
 });
 
-function connect(callback) {
-    connection.connect(function (error) {
+function addUser(idm_username, username, password, hdfsQuota, callback) {
+    pool.getConnection(function(error, connection) {
         if (error) {
             callback(error);
         } else {
-            logger.info('Connected to http://' + mysqlConfig.host + ':' + mysqlConfig.port + '/' +
-                mysqlConfig.database);
-            callback(null);
+            var query = connection.query(
+                'INSERT INTO cosmos_user (idm_username, username, password, hdfs_quota) ' +
+                'VALUES (?, ?, ?, ?)',
+                [idm_username, username, password, hdfsQuota],
+                function (error, result) {
+                    if (error) {
+                        callback(error)
+                    } else {
+                        logger.info('Successful insert: \'INSERT INTO cosmos_user ' +
+                            '(idm_username, username, password, hdfs_quota) VALUES ' +
+                            '(' + idm_username + ', ' + username + ', ' + password + ', ' + hdfsQuota + ')\'');
+                        connection.release();
+                        callback(null, result);
+                    } // if else
+                }
+            );
         } // if else
     });
-} // connect
-
-function addUser(idm_username, username, password, hdfsQuota, callback) {
-    var query = connection.query(
-        'INSERT INTO cosmos_user (idm_username, username, password, hdfs_quota) ' +
-        'VALUES (?, ?, ?, ?)',
-        [idm_username, username, password, hdfsQuota],
-        function (error, result) {
-            if (error) {
-                callback(error)
-            } else {
-                logger.info('Successful insert: \'INSERT INTO cosmos_user ' +
-                    '(idm_username, username, password, hdfs_quota) VALUES ' +
-                    '(' + idm_username + ', ' + username + ', ' + password + ', ' + hdfsQuota + ')\'');
-                callback(null, result);
-            } // if else
-        }
-    );
 } // addUser
 
 function addPassword(idm_username, password, callback) {
-    var query = connection.query(
-        'UPDATE cosmos_user SET password=\'' + password + '\' WHERE idm_username=\'' + idm_username + '\'',
-        function(error, result) {
-            if (error) {
-                callback(error);
-            } else {
-                logger.info('Successful update: \'UPDATE cosmos_user SET password=\'' + password +
-                    '\' WHERE idm_username=\'' + idm_username + '\'');
-                callback(null, result);
-            } // if else
-        }
-    );
+    pool.getConnection(function(error, connection) {
+        if (error) {
+            callback(error);
```

Comment:
Since you are exporting the function `addPassword()` which "may" take a callback (the "client" calling it may pass it or not), maybe you would like to protect the calling of the callback with something such as:
```javascript
if (callback) {
  callback();
}
```